src/Microsoft.OData.Core/ODataRawOutputContext.cs

```diff
@@ -186,10 +184,11 @@ internal override Task WriteInStreamErrorAsync(ODataError error, bool includeDeb
         {
             if (this.outputInStreamErrorListener != null)
             {
-                this.outputInStreamErrorListener.OnInStreamError();
+                return this.outputInStreamErrorListener.OnInStreamErrorAsync();

```

Comment:
My guess is that we were not returning earlier due to the limited async support. right?

Reply:
@marabooy Thank you for the question. I just realized that introducing the return statement changes the logic altogether. Funny how I made the mistake... The `outputInStreamErrorListener` variable in this class only gets initialized to an instance of `ODataAsynchronousWriter` and `OnInStreamError` or `OnInStreamErrorAsync` methods defined in that class actually throw an exception such that no statement after calling either method would ever be executed. So I introduced the return statement based on that knowledge! But that is an implementation detail that can change. The right thing is to add the `async` modifier to the method and await `this.outputInStreamErrorListener.OnInStreamErrorAsync()`. That way the exception thrown by `OnInStreamErrorAsync` or by the `throw` statement a couple of lines below will be captured and placed on the returned task.